.devcontainer/devcontainer.json

```
@@ -11,13 +11,29 @@
     },
     "remoteEnv": {
         // Allow X11 apps to run inside the container
-        "DISPLAY": "${localEnv:DISPLAY}"
+        "DISPLAY": "${localEnv:DISPLAY}",
+        // Put things that allow it in the persistent cache
+        "PRE_COMMIT_HOME": "/cache/pre-commit",
+        "UV_CACHE_DIR": "/cache/uv",
+        "UV_PYTHON_CACHE_DIR": "/cache/uv-python",
+        // Make a venv that is specific for this workspace path as the cache is shared
+        "UV_PROJECT_ENVIRONMENT": "/cache/venv-for${localWorkspaceFolder}",
```

nit:
```suggestion
        "UV_PROJECT_ENVIRONMENT": "/cache/venv-for-${localWorkspaceFolder}",
```
otherwise I think the directory becomes `venv-forblueapi` etc